pilot/pkg/serviceregistry/kube/controller/pod.go

```diff
@@ -166,8 +166,15 @@ func (pc *PodCache) getPodByIP(addr string) *v1.Pod {
 	return item.(*v1.Pod)
 }
 
-// getPod loads the pod from k8s.
+// getPod loads will try to load the pod from informer store and if it is not available, load it from k8s.
 func (pc *PodCache) getPod(name string, namespace string) *v1.Pod {
+	key := kube.KeyFunc(name, namespace)
+	// Try loading it from informer cache by key.
```

Comment:
This does not fully solve the deleted pod case mentioned in the issue though. Thinking about it

Reply:
This also won't fix scale up quickly issues if we have a miss - basically once we hit the Pods() call more than a couple times it's all over, we will likely never recover if there is a large churn rate

Reply:
Now added recentDeletedPods to reduce the api calls during scale up/down for deleted pods